src/main/java/org/folio/service/transactions/batch/BatchPaymentCreditService.java

```diff
@@ -76,10 +76,12 @@ private void updateEncumbranceToCancelTransaction(Transaction transaction, Trans
     double amount = encumbrance.getAmount();
     if (transaction.getTransactionType() == PAYMENT) {
       expended = subtractMoney(expended, transaction.getAmount(), currency);
-      amount = sumMoney(amount, transaction.getAmount(), currency);
+      if (encumbrance.getEncumbrance().getStatus() == Encumbrance.Status.UNRELEASED) {
+        amount = sumMoney(amount, transaction.getAmount(), currency);
+      }
     } else {
       credited = subtractMoney(credited, transaction.getAmount(), currency);
-      if (Encumbrance.Status.RELEASED != encumbrance.getEncumbrance().getStatus()) {
+      if (encumbrance.getEncumbrance().getStatus() == Encumbrance.Status.UNRELEASED) {
```

Comment:
the issue was that we updated encumbrance amoun also for  Encumbrance.Status.PENDING?

Reply:
This is the issue this particular line is fixing, yes.